rclcpp/src/rclcpp/typesupport_helpers.cpp

```diff
@@ -0,0 +1,146 @@
+// Copyright 2018, Bosch Software Innovations GmbH.
+// Copyright 2021, Apex.AI Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "rclcpp/typesupport_helpers.hpp"
+
+#include <functional>
+#include <memory>
+#include <sstream>
+#include <stdexcept>
+#include <string>
+#include <tuple>
+#include <utility>
+
+#include "ament_index_cpp/get_package_prefix.hpp"
+#include "ament_index_cpp/get_resources.hpp"
+#include "rcpputils/shared_library.hpp"
+#include "rcpputils/find_library.hpp"
+#include "rosidl_runtime_cpp/message_type_support_decl.hpp"
+
+namespace rclcpp
+{
+
+namespace 
```

Comment:
linter is complaining about trailing whitespace 
```suggestion
namespace
```

Reply:
applied in f5d8d14